gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/SlackController.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.controllers;
+
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.gate.config.SlackConfigProperties;
+import com.netflix.spinnaker.gate.services.SlackService;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import io.swagger.annotations.ApiOperation;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/slack")
+@ConditionalOnProperty(
```

Comment:
I think this is fine since the controller is the only thing depending on `SlackService`.

Another approach would be to annotate this controller with `@ConditionalOnBean(SlackService.class)` then annotate `SlackConfig.slackService` with `@ConditionalOnProperty("slack.....")` but you don't have to do this. (maybe until slackService gets used elsewhere)